tests-e2e/appstudio/webhook_test.go

```diff
@@ -306,6 +329,70 @@ var _ = Describe("Webhook E2E tests", func() {
 
 		})
 
+		It("Should validate update GitOpsDeployment CR Webhooks for empty Environment field with non-empty namespace.", func() {
+
+			if !isWebhookInstalled("gitopsdeployments", k8sClient) {
+				Skip("skipping as gitopsdeployments webhook is not installed")
+			}
+
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+			ctx = context.Background()
+
+			k8sClient, err = fixture.GetE2ETestUserWorkspaceKubeClient()
+			Expect(err).To(Succeed())
+
+			kubeConfigContents, apiServerURL, err := fixture.ExtractKubeConfigValues()
+			Expect(err).To(BeNil())
+
```

Comment:
One thing I didn't see during my first pass of the review was that we probably should put common code into some utility function or utils file.   For example, in the core package, we have https://github.com/redhat-appstudio/managed-gitops/blob/260fa94067f4dd1a19c9599e3007d98eb95f9534/tests-e2e/core/managed_environment_test.go#L1091, which we can probably reuse here.

Don't want to hold up your PR so we can do this separately. I'll open a Jira, and make not of the changes there.

I'll open a Jira for this.

Reply:
sure, thank you

Reply:
I opened https://issues.redhat.com/browse/GITOPSRVCE-671 to track that work

Reply:
thanks @keithchong